<?xml version="1.0"?>
<!--
Software License Agreement (BSD)

\file      fusion.launch
\authors   Pengdi Huang <alualu628628@163.com>
\copyright Copyright (c) 2022, 06, All rights reserved.


THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WAR-
RANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, IN-
DIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->

<launch>

    <arg name="fileoutputpath" default="/home/vcc/Dense_ROS/" />

    <arg name="cloudintopic" default="/processed_clouds" />

    <arg name="meshouttopic" default="/poisson_mesh" />

    <arg name="meshoutid" default="map" />

    <arg name="poissondepth" default="10" />

    <arg name="poissondensity" default="0.1" />

    
    <!-- frames_fusion node in frames_fusion pkg-->
    <node name="poisson_reconstruction" pkg="poisson_reconstruction" type="poisson_reconstruction_py.py">

        <param name="file_outputpath" type="string" value="$(arg fileoutputpath)"  />

        <param name="cloud_in_topic" type="string" value="$(arg cloudintopic)" />

        <param name="mesh_out_topic" type="string" value="$(arg meshouttopic)" />

        <param name="mesh_out_id" type="string" value="$(arg meshoutid)" />

        <param name="poisson_depth" type="int" value="$(arg poissondepth)" />

        <param name="poisson_density" type="double" value="$(arg poissondensity)" />

    </node>


</launch>
